### PR TITLE
add CodePush custom exceptions

### PR DESCRIPTION
**Pull request by sergey-akhalkov**

### CODE REVIEW

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushMalformedDataException.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * Constructor for malformed CodePush data exception that have
+     * <code>path</code> and <code>message</code> arguments.
+     *
+     * @param path the path to the CodePush update contents.
+     * @param cause the cause why CodePush data cannot be parsed.
+     */
+    public CodePushMalformedDataException(String path, Throwable cause) {
+        super("Unable to parse contents of " + path + ", the file may be corrupted.", cause);
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushMalformedDataException.java]: Do we need quotes around the path?

Comment (sergey-akhalkov) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushMalformedDataException.java]: fixed, thanks

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidPublicKeyException.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.microsoft.appcenter.codepush.exceptions;
+
+/**
+ * Exception class for throwing invalid CodePush public key exceptions.
+ */
+public class CodePushInvalidPublicKeyException extends RuntimeException {
+
+    /**
+     * Creates instance of the CodePush public key exception using
```
Comment (annakocheshkova) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidPublicKeyException.java]: It's better to use this wording for constructors (I've fixed).

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushMalformedDataException.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.microsoft.appcenter.codepush.exceptions;
+
+import java.net.MalformedURLException;
+
+/**
+ * Exception class for throwing malformed CodePush data exceptions.
```
Comment (annakocheshkova) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushMalformedDataException.java]: Can we provide more info about what exactly this malformed data is?

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidPublicKeyException.java (hunk 2 above)
Comment (sergey-akhalkov) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidPublicKeyException.java]: Agreed, thanks!

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushMalformedDataException.java (hunk 3 above)
Comment (sergey-akhalkov) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushMalformedDataException.java]: clarified, please verify

**Pull request merged**